generator/karaf/src/main/java/io/fabric8/maven/generator/karaf/KarafGenerator.java

```diff
@@ -30,8 +32,28 @@
 
 public class KarafGenerator extends BaseGenerator {
 
-    private static final String IMAGE_S2I_UPSTREAM_VERSION = "1.3";
-    private static final String IMAGE_S2I_PRODUCT_VERSION = "2.0";
+    private static final String SETTINGS_RESOURCE_FILE = "/META-INF/fabric8/karaf-default-images.properties";
+    private static final Properties BUILD_SETTINGS;
+
+    static {
+        BUILD_SETTINGS = new Properties();
+        try {
+            BUILD_SETTINGS.load(KarafGenerator.class.getResourceAsStream(SETTINGS_RESOURCE_FILE));
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Cannot load default images properties " + SETTINGS_RESOURCE_FILE + ": " + e,e);
+        }
+    }
+
+    private static String getBuildSetting(String key) {
+        String val = BUILD_SETTINGS.getProperty(key);
+        if (val == null) {
+            throw new IllegalArgumentException("Cannot retrieve default value " + key + " from " + SETTINGS_RESOURCE_FILE);
+        }
+        return val;
+    }
+
```

Comment:
Wonder whether we should combine this lookup with the one of the JavaRunGenerator s, too. But we can do this always later, too.